src/AspNetCoreModuleV2/InProcessRequestHandler/dllmain.cpp

```diff
@@ -93,7 +95,14 @@ CreateApplication(
     try
     {
         RETURN_IF_FAILED(InitializeGlobalConfiguration(pServer, pHttpApplication));
-        
+
+        // In process application was already created
+        if (g_fInProcessApplicationCreated)
+        {
+            *ppApplication = new ShuttingDownApplication(*pServer, *pHttpApplication);
```

Comment:
>  Why does a second call to CreateApplication() indicate shutdown?

Because only one InProcess application can be created per w3wp process (CLR is not restartable). If `CreateApplication` being called the second time means that original InProcess application was recycled and server shutdown is happening. So we return the application that would serve "server shut down" messages.

Reply:
Let's add a comment here to avoid this confusion in the future as it is fairly confusing.

Reply:
OK, please add this in a comment.